orchagent/observer.h

```diff
@@ -30,26 +38,64 @@ class Observer
 class Subject
 {
 public:
+    virtual void attach(SubjectType type, Observer *observer)
```

Comment:
why changes to observer class? I can understand the subject types added above but please explain the motivation here as this is a common code and we don't want to break something. @qiluo-msft to review as well.

Reply:
@prsunny , @qiluo-msft : These changes are enhancement to support registering for specific message types. I have removed the code for now reverted the old changes. As suggested we may push these part a separate PR.
Thanks,
-Praveen.